### PR TITLE
[ci skip] CHANGELOG.md: 4.3.00 update to develop

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

File: CHANGELOG.md
Comment (masterleinad) [inline CHANGELOG.md:3]: I noticed that the dates in this file don't really match https://github.com/kokkos/kokkos/releases. Maybe, we just remove those in here in a follow-up?

**Review by masterleinad — APPROVED** [review]

File: CHANGELOG.md
Comment (ndellingwood) [reply CHANGELOG.md:3]: Yeah, the date you marked is from when I created the changelog PR to the rc. I think removing the dates in a follow up (and excluding in following releases) sounds like a good plan, otherwise that needs to be updated right before final approval+merge

**Pull request merged**
